package/MDAnalysis/core/selection.py

```diff
@@ -1353,6 +1353,51 @@ def _apply(self, group):
         return group[mask]
 
 
+class RelPropertySelection(PropertySelection):
+    """Some of the possible properties:
+    x, y, z,
+
+    .. versionadded:: 2.9.0
+    """
+
+    token = "relprop"
+    precedence = 1
+
+    def __init__(self, parser, tokens):
+        super().__init__(parser, tokens)
+        self.sel = parser.parse_expression(self.precedence)
+        # self.ori_value = self.value
+
+    def _apply(self, group):
+        try:
+            values = getattr(group, self.props[self.prop])
+        except KeyError:
+            errmsg = f"Expected one of {list(self.props.keys())}"
+            raise SelectionError(errmsg) from None
+        except NoDataError:
+            attr = self.props[self.prop]
+            errmsg = f"This Universe does not contain {attr} information"
+            raise SelectionError(errmsg) from None
+
+        try:
+            col = {"x": 0, "y": 1, "z": 2}[self.prop]
+        except KeyError:
+            pass
+        else:
+            values = values[:, col]
+            sel = self.sel.apply(group)
+            rel_value = (
+                sel.center_of_geometry().reshape(3).astype(np.float32)[col]
+            )
+            values -= rel_value
```

Comment:
This calculation may not be correct under PBC.

Look into [MDAnalysis.lib.distances.distance_array](https://docs.mdanalysis.org/dev/documentation_pages/lib/distances.html#MDAnalysis.lib.distances.distance_array) (or probably better [MDAnalysis.lib.distances.capped_distance](https://docs.mdanalysis.org/dev/documentation_pages/lib/distances.html#MDAnalysis.lib.distances.capped_distance)) and [MDAnalysis.lib.distances.minimize_vectors](https://docs.mdanalysis.org/dev/documentation_pages/lib/distances.html#MDAnalysis.lib.distances.minimize_vectors) for fast functions that can calculate distances under PBC.